### PR TITLE
MOIA-28311: Update to AWS Java SDK v2

### PR DESCRIPTION
**Pull request by saeltz**
Updates to the AWS Java SDK v2
- Remove outdated SignableHttpRequest

### CODE REVIEW

**Comment by saeltz** [comment]
Scala 2.12 support makes the whole code much more complicated and so much worse.
After some research, only `trip-estimate` and `trip-booking2` seem to use this and both use Scala 2.13. So we could drop support for Scala 2.12 and release a new major version.
What do you think?

**Comment by saeltz** [comment]
Now only building for Scala 2.13. This requires a new major release.

File: src/test/scala/io/moia/scalaHttpClient/AwsRequestSignerTest.scala [hunk 1]
```diff
@@ -14,20 +13,42 @@ class AwsRequestSignerTest extends AnyWordSpecLike with Matchers with FutureValu
   protected implicit val patienceConfig: FiniteDuration = 2.seconds
 
   classOf[AwsRequestSigner].getSimpleName should {
-    "sign http requests" in {
-      def dateHeader: HttpHeader =
-        HttpHeader.parse("X-Amz-Date", "20150830T123600Z") match {
-          case Ok(header, _) => header
-          case unexpected    => sys.error(s"Unexpected HTTP header parse result: $unexpected")
-        }
+    "sign a villa http GET request" in {
```
Comment (JannikArndt) [inline src/test/scala/io/moia/scalaHttpClient/AwsRequestSignerTest.scala:16]: Whats a "villa http"? 😄

Comment (JannikArndt) [reply src/test/scala/io/moia/scalaHttpClient/AwsRequestSignerTest.scala:16]: Aaahhhh
```suggestion
    "sign a vanilla http GET request" in {
```

Comment (JannikArndt) [reply src/test/scala/io/moia/scalaHttpClient/AwsRequestSignerTest.scala]: I rephrased it in 8950457

**Pull request merged**